### PR TITLE
Support branch only builds with Github Branch Source Plugin

### PR DESCRIPTION
**Pull request by jayv**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -50,10 +50,10 @@
     </issueManagement>
 
     <properties>
-        <jenkins.version>1.580.1</jenkins.version>
-        <!--<jenkins.version>2.46.3</jenkins.version>-->
-        <java.level>6</java.level>
-        <!--<java.level>7</java.level>-->
+        <jenkins.version>2.60.2</jenkins.version>
```
Comment (terma) [inline pom.xml]: Please do not increase version, because that's remove support of old version of Jenkins.

File: pom.xml [hunk 2]
```diff
@@ -50,10 +50,10 @@
...
+        <jenkins-test-harness.version>2.23</jenkins-test-harness.version>
+        <java.level>8</java.level>
```
Comment (terma) [inline pom.xml]: I like Java 8 as well but old version of Jenkins could be runned on any Java so we don't want to stop support them.

File: pom.xml [hunk 3]
```diff
@@ -87,7 +87,7 @@
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>github-api</artifactId>
-            <version>1.71</version>
+            <version>1.86</version>
```
Comment (terma) [inline pom.xml]: Is there any reason why you need to increase version?

File: src/main/java/com/github/terma/jenkins/githubprcoveragestatus/CompareCoverageAction.java [hunk 4]
```diff
@@ -74,10 +87,29 @@ public void perform(
...
         }
 
+
         buildLog.println(BUILD_LOG_PREFIX + "start");
 
-        final String gitUrl = Utils.getGitUrl(build, listener);
-        final int prId = Utils.gitPrId(build, listener);
+        String gitUrl;
+        int prId;
+        try {
```
Comment (terma) [inline src/main/java/com/github/terma/jenkins/githubprcoveragestatus/CompareCoverageAction.java]: Please replace that ```try catch``` logic on simple ```if``` for example:
```
if (Utils.getPrId(build, listener) != null) // logic
else if (scmVars != null && scmVars.contains(...)) // logic
else throw new IAE("Please provide PR_ID or scmVars");
``` 
That's simplify logic and you can avoid additional settings ```isPrDiscoveryForBranches```

File: pom.xml (hunk 1 above)
Comment (jayv) [inline pom.xml]: Cool, I don't think the code depends on anything new, but for my testing wanted to align it with our Jenkins version.

File: pom.xml (hunk 3 above)
Comment (jayv) [inline pom.xml]: I think that's a 2.60 dependency as well, but I'll revert and see if I can get a build working without this.

File: src/main/java/com/github/terma/jenkins/githubprcoveragestatus/CompareCoverageAction.java (hunk 4 above)
Comment (jayv) [inline src/main/java/com/github/terma/jenkins/githubprcoveragestatus/CompareCoverageAction.java]: `Utils.get*` throws exceptions in all cases, so... I dislike try/catch for control logic just as much, but didn't want to change too much, do we want to refactor `Utils` to return `null`?

I agree on the removing the setting, passing `scmVars` implies you opt-in to branch discovery.

**Comment by jayv** [comment]
Reverted dependencies and tested against `2.60.3` everything still works fine, so if you can suggest how you'd like to handle the `Utils` refactoring I'll work on that and add/update some tests for the new use-cases.

**Comment by terma** [comment]
Implemented in scope of #36

**Pull request closed without merging**